blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,657 @@
+	"interfaces": [
+		{
+			"name": "eth0",
+			"ipAddresses": [
+				{
+					"address": "192.168.0.0/32",
+					"serviceAddress": true
+				},
+				{
+					"address": "dead::babe/128",
+					"serviceAddress": true
+				}
+			]
+		}
+	],
+	"port": 80,
+	"profile": "ATS_EDGE",
+	"status": "REPORTED",
+	"type": "EDGE"
+}
+```
+Note that only interfaces that have "monitor" set to `true` should appear at all
+in the payloads of this endpoint. Further note that neither gateways nor
+<abbr title="Maximum Transmission Unit">MTU</abbr> are reported.
+
+##### `/cdns/{{name}}/snapshot`
+This is a semantic change only - payloads remain structurally unchanged. The
+reported `ip`, `ip6` and `interfaceName` fields for elements of the
+`contentServers` array property should be an arbitrarily chosen IPv4 service
+address, an arbitrarily chosen IPv6 service address, and the interface name of
+that/those chosen address(es).
+
+##### `/cdns/{{name}}/snapshot/new`
+This is a semantic change only - payloads remain structurally unchanged. The
+changes to this endpoint are the same as those described for
+`/cdns/{{name}}/snapshot` above.
+
+##### `/servers`/`/servers/{{ID}}`
+An example of one element of an output array for this endpoint at the time of
+this writing is shown below.
+```json
+{
+	"cachegroup": "CDN_in_a_Box_Mid",
+	"cachegroupId": 6,
+	"cdnId": 2,
+	"cdnName": "CDN-in-a-Box",
+	"domainName": "infra.ciab.test",
+	"guid": null,
+	"hostName": "mid",
+	"httpsPort": 443,
+	"id": 10,
+	"iloIpAddress": "",
+	"iloIpGateway": "",
+	"iloIpNetmask": "",
+	"iloPassword": "",
+	"iloUsername": "",
+	"interfaceMtu": 1500,
+	"interfaceName": "eth0",
+	"ip6Address": "fc01:9400:1000:8::120",
+	"ip6Gateway": "fc01:9400:1000:8::1",
+	"ipAddress": "172.16.239.120",
+	"ipGateway": "172.16.239.1",
+	"ipNetmask": "255.255.255.0",
+	"lastUpdated": "2018-12-05 18:45:05+00",
+	"mgmtIpAddress": "",
+	"mgmtIpGateway": "",
+	"mgmtIpNetmask": "",
+	"offlineReason": "",
+	"physLocation": "Apachecon North America 2018",
+	"physLocationId": 1,
+	"profile": "ATS_MID_TIER_CACHE",
+	"profileDesc": "Mid Cache - Apache Traffic Server",
+	"profileId": 10,
+	"rack": "",
+	"revalPending": false,
+	"routerHostName": "",
+	"routerPortName": "",
+	"status": "REPORTED",
+	"statusId": 3,
+	"tcpPort": 80,
+	"type": "MID",
+	"typeId": 12,
+	"updPending": false,
+	"xmppId": "mid",
+	"xmppPasswd": "",
+	"ipIsService": true,
+	"ip6IsService": true
+}
+```
+After this feature's inclusion, the same element would look like this:
+```json
+{
+	"cachegroup": "CDN_in_a_Box_Mid",
+	"cachegroupId": 6,
+	"cdnId": 2,
+	"cdnName": "CDN-in-a-Box",
+	"domainName": "infra.ciab.test",
+	"guid": null,
+	"hostName": "mid",
+	"httpsPort": 443,
+	"id": 10,
+	"iloIpAddress": "",
+	"iloIpGateway": "",
+	"iloIpNetmask": "",
+	"iloPassword": "",
+	"iloUsername": "",
+	"interfaces": [
+		{
+			"ipAddresses": [
+				{
+					"address": "172.16.239.120/24",
+					"gateway": "172.16.239.1",
+					"serviceAddress": true
```

Comment:
just wondering. is `serviceAddress` applicable to all interfaces? for the interface that is included in the TR snapshot, this has meaning but does it have meaning for the other interfaces?

Reply:
From a different point in the blueprint:

> _"A server object should be accepted (based only on its `interfaces` property) if it has at least one interface, every interface has at least one IP address, exactly one interface has at least one IP address that is a service address, and no two IP addresses of the same family/version are simultaneously marked as 'service addresses'."_

So to directly answer your question, only one interface on a server object is allowed to have ipAddresses that are "service addresses". The meaning that field has for other, non-"active" interfaces is that its value being `true` indicates an erroneous configuration, because only the "active" interface has "service addresses", and in fact the "active" interface, as you put it, is defined by having said "service addresses".